### ISSUE TITLE
New noöp initContainer Rejected in Namespaces that have ResourceQuota

### ISSUE THREAD

**Issue by ripta** [linkerd/linkerd2#10162]
### What is the issue?

We attempted to upgrade to the latest stable release (linkerd stable-2.12.3), and ran into a problem with the new noöp container in namespaces that have resource quota, because the noöp init container does not have resource limits defined on it.

We run `proxy-injector`, and I couldn't figure out a good way of adding resource limits / requests without basically modifying the `patch.json` in the chart and recompiling the injector.

### How can it be reproduced?

Create a new namespace and a resource quota in the namespace:

```
apiVersion: v1
kind: Namespace
metadata:
  name: d-ripta-examples--emojivoto

---
apiVersion: v1
kind: ResourceQuota
metadata:
  name: workspace-compute-resources
  namespace: d-ripta-examples--emojivoto
spec:
  hard:
    limits.cpu: 40
    limits.memory: 20Gi
```

Deploy an app that has `linkerd.io/inject: enabled` on it. I'm using emojivoto. The deployment applies, but replicasets never make progress.

### Logs, error output, etc

Viewing events (`kc -n $namespace get events`), we see:

```
Error creating: pods "emoji-66f8f788bd-bgnz9" is forbidden: failed quota: workspace-compute-resources: must specify limits.cpu,limits.memory
```

### output of `linkerd check -o short`

-

### Environment

- Kubernetes 1.22, linkerd stable-2.12.3

### Possible solution

We can temporarily create `LimitRange` to assign defaults:

```
apiVersion: v1
kind: LimitRange
metadata:
  name: linkerd-default-for-init-container
  namespace: d-ripta-examples--emojivoto
spec:
  limits:
  - default:
      cpu: 10m
      memory: 50Mi
    defaultRequest:
      cpu: 10m
      memory: 50Mi
    type: Container
```

but would prefer to be able to configure sane defaults in our helm config.

### Additional context

I'm aware that the noöp container was replaced with network-validator in linkerd edge, but the same problem will most likely manifest itself as the network-validator partial is also missing resource limits.

### Would you like to work on fixing this bug?

None

**Comment by kleimkuhler** [comment]
Closed by #10253 (2.12 patch) and #10254 (2.13).

**Comment by jeremychase** [comment]
We want to pipe through the proxInit helm values for resource requests/limits to resolve the `noop` container hitting the limit: https://github.com/linkerd/linkerd2/pull/10253#issuecomment-1439625136
